### PR TITLE
DROOLS-2299: Relation Editor inline controls

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/DROOLS-2299

Context Menu for ```Relation``` plus support for deleting ```ContextEntry``` rows (missing from #1428)

This is the last of the "old" requirement developments that were WIP.. now to circle back and redo them.

See video [here](https://youtu.be/wZMMVx96erQ).

### CODE REVIEW

**Comment by manstis** [comment]
Jenkins please retest this.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationColumnCommand.java [hunk 1]
```diff
@@ -37,55 +39,42 @@
...
-            if (cell == null) {
-                cells.add(new NullGridCell());
-            } else {
-                cells.add(cell);
-            }
-        }
-        return cells;
+    private List<Expression> extractColumnData(final int uiColumnIndex) {
+        final int iiIndex = uiColumnIndex - RelationUIModelMapperHelper.ROW_INDEX_COLUMN_COUNT;
+        final List<Expression> values = relation.getRow().stream().map(row -> row.getExpression().get(iiIndex)).collect(Collectors.toList());
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationColumnCommand.java]: Do we have this local variable `values` for debugging purposes? If not, we could shorten the statement:
```java
return relation.getRow().stream().map(row -> row.getExpression().get(iiIndex)).collect(Collectors.toList());
```

Let me know WDYT.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationColumnCommandTest.java [hunk 2]
```diff
@@ -152,6 +171,10 @@ public void testGraphCommandUndoWithRows() {
                      relation.getColumn().size());
         assertEquals(1,
                      relation.getRow().size());
+        assertEquals(1,
+                     relation.getRow().get(0).getExpression().size());
+        assertEquals("value",
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationColumnCommandTest.java]: Maybe we could extract "value" to local variable.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationColumnCommandTest.java [hunk 3]
```diff
@@ -233,41 +257,10 @@ public void testCanvasCommandExecuteWithNoRows() {
     @Test
     public void testCanvasCommandUndoWithRows() {
         relation.getRow().add(new List());
+        relation.getRow().get(0).getExpression().add(new LiteralExpression());
+        ((LiteralExpression) relation.getRow().get(0).getExpression().get(0)).setText("value");
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationColumnCommandTest.java]: This line is hard readable, I think if we rewrote two lines above, we could get nicer code:
```java
final LiteralExpression literalExpression = new LiteralExpression();
literalExpression.setText("value");
relation.getRow().get(0).getExpression().add(literalExpression);
```

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationRowCommandTest.java [hunk 4]
```diff
@@ -123,6 +139,8 @@ public void testGraphCommandExecuteWithNoColumns() {
     @Test
     public void testGraphCommandUndoWithColumns() {
         relation.getColumn().add(new InformationItem());
+        relation.getRow().get(0).getExpression().add(new LiteralExpression());
+        ((LiteralExpression) relation.getRow().get(0).getExpression().get(0)).setText("value");
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationRowCommandTest.java]: Similar readability comment here.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationRowCommandTest.java [hunk 5]
```diff
@@ -204,7 +226,12 @@ public void testCanvasCommandExecuteWithNoColumns() {
 
     @Test
     public void testCanvasCommandUndoWithColumns() {
+        relation.getColumn().add(new InformationItem());
+        relation.getRow().get(0).getExpression().add(new LiteralExpression());
+        ((LiteralExpression) relation.getRow().get(0).getExpression().get(0)).setText("value");
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationRowCommandTest.java]: And here.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationColumnCommand.java (hunk 1 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationColumnCommand.java]: Yes it was for that reason.. will change now it's "working"!

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationColumnCommandTest.java (hunk 2 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationColumnCommandTest.java]: OK. Will do.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationColumnCommandTest.java (hunk 3 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationColumnCommandTest.java]: ... and there you spot my laziness! Will make more readable as you suggest.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationRowCommandTest.java (hunk 4 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationRowCommandTest.java]: Ditto.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationRowCommandTest.java (hunk 5 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/expressions/types/relation/DeleteRelationRowCommandTest.java]: Ditto.

**Comment by manstis** [comment]
@jomarko You need to rebase this PR with master to build locally (however that'll be done in my next commit containing the fixes to the items you raised above).

**Comment by manstis** [comment]
@jomarko Updates made.

**Comment by manstis** [comment]
Jenkins please retest this.

**Pull request merged**